### ISSUE TITLE
ValueError: array is too big; `arr.size * arr.dtype.itemsize` is larger than the maximum possible size.

### ISSUE THREAD

**Issue by EveTC** [isaacovercast/easySFS#29]
Hello,

I have a vcf file with 522 indidividuals from 30 populations. The SNPs were called using freebayes + samtools. I have filtered them using vcftools. I am hoping to get a SFS for this dataset.

I followed the steps suggested and use the `--preview` to decide on the number to use for each population. I then ran the `--proj` using thes numbers. It output as far as I can tell all the 1D and 2D sfs. However I recieve this error:
`ValueError: array is too big; `arr.size * arr.dtype.itemsize` is larger than the maximum possible size.`
whilst it tries to make the multiSFS.

The full error is:
```
Doing multiSFS for all pops
Traceback (most recent call last):
  File "/pub37/eve/bin/miniconda3/envs/Genomics_processing/easySFS/easySFS.py", line 650, in <module>
    main()
  File "/pub37/eve/bin/miniconda3/envs/Genomics_processing/easySFS/easySFS.py", line 635, in main
    sfs_file = dadi_multiSFS(dd, pops, proj=proj, unfold=args.unfolded, outdir=outdir, prefix=prefix, dtype=args.dtype)
  File "/pub37/eve/bin/miniconda3/envs/Genomics_processing/easySFS/easySFS.py", line 156, in dadi_multiSFS
    fs = dadi.Spectrum.from_data_dict(dd, pops, proj, polarized=unfold)
  File "/pub37/eve/bin/miniconda3/envs/Genomics_processing/lib/python3.7/site-packages/dadi/Spectrum_mod.py", line 1848, in from_data_dict
    fs = Spectrum._from_count_dict(cd, projections, polarized, pop_ids)
  File "/pub37/eve/bin/miniconda3/envs/Genomics_processing/lib/python3.7/site-packages/dadi/Spectrum_mod.py", line 1874, in _from_count_dict
    fs_total = dadi.Spectrum(numpy.zeros(numpy.array(projections)+1),
ValueError: array is too big; `arr.size * arr.dtype.itemsize` is larger than the maximum possible size.
```

Is there a way to increase the max possible size? If you need more info please let me know.
Thank you for your help.
Eve

**Comment by EveTC** [comment]
Hi Isaac,

Haha yes - I think I got a bit over excited with my sampling! Having a second think about it I may be able to combine some of the populations based on structure analysis - so that will hopefully reduce the total number of popualtions.

My `--proj` values range from 6-34.
For future reference - do you know what the max number of populations or individuals that EasySFS can manage?

Thank you so much for you help.
Also thank you for the package it seems like it will be super useful! :)

**Comment by isaacovercast** [comment]
Hello, yeah I don't know the max number of pops or individuals. Really this limit is imposed by numpy, so the size of the internal matrix generated by dadi. My thought on this is if you are _anywhere_ near this size limit any downstream analysis will take effectively forever to run. A "typical" large multi sfs is less than 5 populations, maybe 10 to 20 individuals. In practice you tend not to see sfs in published analyses with more than 3 or 4 populations.

In short, the max number of populations will be much more tightly constrained by whatever analysis you're doing downstream, than by easySFS.
